src/hb-ot-var-common.hh

```diff
@@ -416,6 +418,45 @@ struct TupleVariationHeader
   DEFINE_SIZE_MIN (4);
 };
 
+/* not using hb_bytes_t: avoid potential build issues with some compilers */
+struct byte_data_t
+{
+  const char *p = nullptr; //allocated by hb_calloc () or similar
+  unsigned length = 0;
+
+  byte_data_t () = default;
+  byte_data_t (const byte_data_t&) = default;
+  byte_data_t (const char *p_, unsigned len_) : p (p_), length (len_) {}
+  ~byte_data_t () = default;
+  byte_data_t& operator= (const byte_data_t&) = default;
+  byte_data_t& operator= (byte_data_t&&) = default;
+
+  void fini () { hb_free ((char *) p); p = nullptr; length = 0; }
+
+  bool operator == (const byte_data_t& o) const
+  { return p == o.p && length == o.length; }
```

Comment:
Should equals compare the bytes instead of the pointer? If you switch to using a hb_bytes_t instance then you can use the == method from hb_bytes_t which should do this. Also I believe we'll need a hash() method if this is to be used as a hashmap key.

Reply:
My thought is to allocate bytes once, and free them only during destruction. Copy and assignment operations of the struct just copy the pointer value, so I think equals doesn't need to compare the bytes, maybe just check the pointer is enough in this case?

Reply:
Are you planning to use these as hash map keys in future changes? If you are then we'll need to the value comparison equals. If they are only planned to be used as hashmap values then this equals is fine.

Reply:
no, they're just used as hashmap values